files/nrpe/check_openstack.py

```diff
@@ -672,6 +681,53 @@ def execute(self):
     except:
       raise
 
+class OSL3Agent():
+  '''
+  Check all Neutron L3 agents are alive
+  '''
+
+  options = dict()
+
+  def __init__(self, options):
+    self.neutron = neutronclient.Client('2', session=keystone_session_v3(options))
+
+  def check_bad_l3_agents(self):
+    err_l3agents = []
+    l3agents = self.neutron.list_agents(agent_type='L3 agent')
+
+    if str(l3agents.keys()) == "[u'agents']":
+      for item in l3agents.values():
+        for x in item:
+          if x['alive'] == True and x['admin_state_up'] == True:
+            OK = 1
+          elif x['alive'] == False and x['admin_state_up'] == False:
+            WARNING = 1
+            err_l3agents.append("%s (admin_state_up=%s, alive=%s) on %s" % (x['binary'],x['admin_state_up'],x['alive'],x['host']))
+          elif x['alive'] == False and x['admin_state_up'] == True:
+            CRITICAL = 1
+            err_l3agents.append("%s (admin_state_up=%s, alive=%s) on %s" % (x['binary'],x['admin_state_up'],x['alive'],x['host']))
+          else:
+            UNKNOWN = 1
+            err_l3agents.append("%s (admin_state_up=%s, alive=%s) on %s" % (x['binary'],x['admin_state_up'],x['alive'],x['host']))
+    else:
+      UNKNOWN = 1
+      err_l3agents = l3agents
```

Comment:
This seems to me unnecessary. Since we are not using l3agents anymore later in it's scope, we can use that to raise the exception.

Reply:
Fixed